src/SelfServe/SelfServeTypes.ts

```diff
@@ -68,27 +68,37 @@ export abstract class SelfServeBaseClass {
     baselineValues: ReadonlyMap<string, SmartUiInput>
   ) => Promise<OnSaveResult>;
   public abstract onRefresh: () => Promise<RefreshResult>;
+  private className: string;
+
+  constructor() {
+    this.className = this.constructor.name;
+  }
+
+  public getClassName(): string {
```

Comment:
Getter methods around private state isn't really idiomatic in TS. I would just use `this.constructor.name` everywhere

Reply:
Will remove that.